cli/flox/src/utils/metrics.rs

```diff
@@ -363,24 +277,670 @@ fn read_metrics_uuid(config: &Config) -> Result<Uuid> {
         })
 }
 
-fn add_metric(event: MetricEvent) -> Result<()> {
-    let config = Config::parse()?;
+static METRICS_HUB: Lazy<Hub> = Lazy::new(|| Hub {
+    client: Arc::new(Mutex::new(None)),
+});
+
+/// A sharable wrapper around a metrics [Client],
+/// that allows for registering and flushing metric events.
+///
+/// A global [Hub] is used by the [MetricsLayer] to record metrics
+/// from tracing events emitted by [subcommand_metric!].
+#[derive(Debug)]
+pub struct Hub {
+    client: Arc<Mutex<Option<Client>>>,
```

Comment:
nonblocking: would it be possible to make this a `Client` instead of an `Option<Client>`? Seems like there's a good chunk of essentially dead code, and you always want either an actual client or a test client. Maybe an `Option` was necessary for testing?

Reply:
We cannot globally initialize this without a defined "uninitialized" or "disabled" state, since the initialization/configuration of the client happens later.
Here, `None` is serving as that state. That leaves the no client set, i.e. metrics disabled checks to the Hub implmentation.
I considered having a noop `Connection` implementation, but that would still capture and write metrics (just never actually send them).